services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightSqlProducer.java

```diff
@@ -122,7 +185,19 @@ public void getStreamPreparedStatement(
     CommandPreparedStatementQuery commandPreparedStatementQuery,
     CallContext callContext, Ticket ticket,
     ServerStreamListener serverStreamListener) {
-    throw CallStatus.UNIMPLEMENTED.withDescription("PreparedStatement not supported.").toRuntimeException();
+    final CallHeaders headers = retrieveHeadersFromCallContext(callContext);
+    final UserSession session = sessionsManager.getUserSession(callContext.peerIdentity(), headers);
+
+    try {
+      UserProtos.PreparedStatementHandle preparedStatementHandle =
+        UserProtos.PreparedStatementHandle.parseFrom(commandPreparedStatementQuery.getPreparedStatementHandle());
```

Comment:
Do you not need to validate that the prepared statement handle is in the cache?

Reply:
Fixed